### ISSUE TITLE
Some robots to improve our workflow?

### ISSUE THREAD

**Issue by mathpluscode** [DeepRegNet/DeepReg#461]
## Subject of the issue

If we have some robots helping us label issues / PRs etc, this might be fun?

I found the following apps
- https://github.com/marketplace/stale, this can notify us in case some issues have been there for too long ;)
- https://github.com/marketplace/dependabot-preview seems that we have this already
- https://github.com/marketplace/wip WIP, that's good to have ;) gitlab has this feature by default.

and actions
- https://github.com/marketplace/actions/build-and-push-docker-images we have dockerfile now, and we should push the image to the docker hub ;) as monai
- https://github.com/actions/starter-workflows/blob/08ff79c7b930315861d16bed4903ddd90bf7c05c/ci/python-publish.yml for release the package to pypi, as in issue #408

**Comment by mathpluscode** [comment]
I checked all verified app from github marketplace and above are the one I've found which might be interesting/fun to have ^^.

What do you think? @ebonmati @zacbaum @s-sd @NMontanaBrown @acasamitjana @RemiDelaunay @YipengHu @sbb-gh 

I didn't tag everyone but feel free to tag others.

**Comment by YipengHu** [comment]
I'm all for robots. Any kind do docs? 😅

**Comment by mathpluscode** [comment]
> I'm all for robots. Any kind do docs? 😅

docs? what docs? @YipengHu

**Comment by zacbaum** [comment]
> ## Subject of the issue
> If we have some robots helping us label issues / PRs etc, this might be fun?
> 
> I found the following apps
> 
> * https://github.com/marketplace/stale, this can notify us in case some issues have been there for too long ;)
> * https://github.com/marketplace/dependabot-preview seems that we have this already
> * https://github.com/marketplace/wip WIP, that's good to have ;) gitlab has this feature by default.

For tickets in progress, we can just ask those (if they are actually actively working on it) to tag it as accepted - just need to create a new tag for that. Though right now, it seems like things aren't as active so it's not much of an issue.

> 
> and actions
> 
> * https://github.com/marketplace/actions/build-and-push-docker-images we have dockerfile now, and we should push the image to the docker hub ;) as monai
> * https://github.com/actions/starter-workflows/blob/08ff79c7b930315861d16bed4903ddd90bf7c05c/ci/python-publish.yml for release the package to pypi, as in issue #291

For auto-pypi releases, we will likely go with something like you've put here with a GitHub action, but I think that's been deemed low-priority for time being, at least until JOSS review is finished.

**Comment by mathpluscode** [comment]
> > ## Subject of the issue
> > If we have some robots helping us label issues / PRs etc, this might be fun?
> > I found the following apps
> > 
> > * https://github.com/marketplace/stale, this can notify us in case some issues have been there for too long ;)
> > * https://github.com/marketplace/dependabot-preview seems that we have this already
> > * https://github.com/marketplace/wip WIP, that's good to have ;) gitlab has this feature by default.
> 
> For tickets in progress, we can just ask those (if they are actually actively working on it) to tag it as accepted - just need to create a new tag for that. Though right now, it seems like things aren't as active so it's not much of an issue.
> 
> > and actions
> > 
> > * https://github.com/marketplace/actions/build-and-push-docker-images we have dockerfile now, and we should push the image to the docker hub ;) as monai
> > * https://github.com/actions/starter-workflows/blob/08ff79c7b930315861d16bed4903ddd90bf7c05c/ci/python-publish.yml for release the package to pypi, as in issue #291
> 
> For auto-pypi releases, we will likely go with something like you've put here with a GitHub action, but I think that's been deemed low-priority for time being, at least until JOSS review is finished.

@zacbaum what do you mean "tag it as accepted"?

oh for the pypi, the current "stable" version is actually buggy as you may see in some issues like #442 #452. Therefore, it'd be better to release a `0.1.0b2` soon, but we can still do it manually.

So what's your general opinion to have some "robots" around so that we don't do things manually? As scripts can be stand-by 24/7 ;)

**Comment by zacbaum** [comment]
> > > ## Subject of the issue
> > > If we have some robots helping us label issues / PRs etc, this might be fun?
> > > I found the following apps
> > > 
> > > * https://github.com/marketplace/stale, this can notify us in case some issues have been there for too long ;)
> > > * https://github.com/marketplace/dependabot-preview seems that we have this already
> > > * https://github.com/marketplace/wip WIP, that's good to have ;) gitlab has this feature by default.
> > 
> > 
> > For tickets in progress, we can just ask those (if they are actually actively working on it) to tag it as accepted - just need to create a new tag for that. Though right now, it seems like things aren't as active so it's not much of an issue.
> > > and actions
> > > 
> > > * https://github.com/marketplace/actions/build-and-push-docker-images we have dockerfile now, and we should push the image to the docker hub ;) as monai
> > > * https://github.com/actions/starter-workflows/blob/08ff79c7b930315861d16bed4903ddd90bf7c05c/ci/python-publish.yml for release the package to pypi, as in issue #291
> > 
> > 
> > For auto-pypi releases, we will likely go with something like you've put here with a GitHub action, but I think that's been deemed low-priority for time being, at least until JOSS review is finished.
> 
> @zacbaum what do you mean "tag it as accepted"?
 
In the same way that we have "bug" and "docs" labels in the issue tracker, we simply have a label that is "accepted" and indicates that the issue is actively being worked on.

> oh for the pypi, the current "stable" version is actually buggy as you may see in some issues like #442 #452. Therefore, it'd be better to release a `0.1.0b2` soon, but we can still do it manually.

Fair point - are these bugs blocking/breaking for major parts of the code? Otherwise, this is likely always the case that bugs will be found and we'll need *another* release 😉 

> So what's your general opinion to have some "robots" around so that we don't do things manually? As scripts can be stand-by 24/7 ;)

As above, not against for releases, but I think automating labels for issues beyond what's already done may be unnecessary and add complexity.

**Comment by mathpluscode** [comment]
@zacbaum Oh, there might be a bit of misunderstanding, the only automated label is "stale" which will remind us when an issue/PR has been inactive for e.g. 30 days. The other labels like "accepted" "confirmed" "priority", yes ofc they gonna be used manually ^^ So what do you think about this "stale", I found it widely used in other OS projects.

For the bugs, I'd consider them as important as for example, using gmi loss might break the code. We can discuss this new beta release in details (what to include in the release) in the next meeting.

**Comment by mathpluscode** [comment]
side note: we need to have a release note / change log manually, I will add a new issue.

**Comment by mathpluscode** [comment]
@zacbaum for the pypi release, do we have a separate issue?

It seems that you have lots of work to do already lol, tell us if you need help ^^.

**Comment by zacbaum** [comment]
> @zacbaum for the pypi release, do we have a separate issue?
> 
> It seems that you have lots of work to do already lol, tell us if you need help ^^.

See #408 - Just need to decide what the priorities are for what I've been assigned... Seems this wasn't one until today so I will get to this after #324.

**Comment by mathpluscode** [comment]
Closing this as the mentioned robots (WIP, stale) have been added, and we have a separate issue for pypi.
